mock.go

```diff
@@ -78,26 +135,146 @@ func (t *TerminalMock) GetResult() string {
-			ek := event.(*tcell.EventKey)
-			t.InjectKey(ek.Key(), ek.Rune(), ek.Modifiers())
-		case *tcell.EventResize:
-			er := event.(*tcell.EventResize)
-			w, h := er.Size()
-			t.SetSize(w, h)
+func (m *TerminalMock) init() error {
+	return nil
+}
+
+func (m *TerminalMock) size() (width int, height int) {
+	m.sizeMu.RLock()
+	defer m.sizeMu.RUnlock()
+	return m.width, m.height
+}
+
+func (m *TerminalMock) clear(fg termbox.Attribute, bg termbox.Attribute) error {
+	// TODO
+	return nil
+}
+
+func (m *TerminalMock) setCell(x int, y int, ch rune, fg termbox.Attribute, bg termbox.Attribute) {
+	m.sizeMu.RLock()
+	defer m.sizeMu.RUnlock()
+	m.cellsMu.Lock()
+	defer m.cellsMu.Unlock()
+
+	if x < 0 || x >= m.width {
+		return
+	}
+	if y < 0 || y >= m.height {
+		return
+	}
+	m.cells[y*m.width+x] = &cell{ch: ch, fg: fg, bg: bg}
+}
+
+func (m *TerminalMock) setCursor(x int, y int) {
+	m.sizeMu.RLock()
+	defer m.sizeMu.RUnlock()
+	m.cellsMu.Lock()
+	defer m.cellsMu.Unlock()
+	if x < 0 || x >= m.width {
+		return
+	}
+	if y < 0 || y >= m.height {
+		return
+	}
+	i := y*m.width + x
+	if m.cells[i] == nil {
+		m.cells[y*m.width+x] = &cell{ch: '\u2588', fg: termbox.ColorWhite, bg: termbox.ColorDefault}
+	} else {
+		// Cursor on a rune.
+		m.cells[y*m.width+x].bg = termbox.ColorWhite
+	}
+}
+
+func (m *TerminalMock) pollEvent() termbox.Event {
+	m.eventsMu.Lock()
+	defer m.eventsMu.Unlock()
+	if len(m.events) == 0 {
+		panic("pollEvent called with empty events. have you set expected events by SetEvents?")
+	}
+	e := m.events[0]
+	m.events = m.events[1:]
+	// Wait a moment for goroutine scheduling.
+	time.Sleep(m.sleepDuration)
+	return e
+}
+
+// flush displays all items with formatted layout.
+func (m *TerminalMock) flush() error {
+	m.cellsMu.RLock()
+
+	var s string
+	for j := 0; j < m.height; j++ {
+		prevFg, prevBg := termbox.ColorDefault, termbox.ColorDefault
+		for i := 0; i < m.width; i++ {
+			c := m.cells[j*m.width+i]
+			if c == nil {
+				s += " "
+				prevFg, prevBg = termbox.ColorDefault, termbox.ColorDefault
+				continue
+			} else {
+				var fgReset bool
+				if c.fg != prevFg {
+					s += "\x1b\x5b\x6d" // Reset previous color.
+					s += parseAttr(c.fg, true)
+					prevFg = c.fg
+					prevBg = termbox.ColorDefault
+					fgReset = true
+				}
+				if c.bg != prevBg {
+					if !fgReset {
+						s += "\x1b\x5b\x6d" // Reset previous color.
+						prevFg = termbox.ColorDefault
+					}
+					s += parseAttr(c.bg, false)
+					prevBg = c.bg
+				}
+				s += string(c.ch)
+				rw := runewidth.RuneWidth(c.ch)
+				if rw != 0 {
+					i += rw - 1
+				}
+			}
 		}
+		s += "\n"
 	}
+	s += "\x1b\x5b\x6d" // Reset previous color.
+
+	m.cellsMu.RUnlock()
+	m.cellsMu.Lock()
+	m.cells = make([]*cell, m.width*m.height)
+	m.cellsMu.Unlock()
+
+	m.resultMu.Lock()
+	defer m.resultMu.Unlock()
+	m.result = s
+
+	return nil
+}
+
+func (m *TerminalMock) close() {}
+
+// UseMockedTerminal switches the terminal, which is used from
+// this package to a mocked one.
```

Comment:
Could you add a comment for deprecation? (ref. https://github.com/golang/go/wiki/Deprecated)

```suggestion
// this package to a mocked one.
//
// Deprecated: Use UseMockedTerminalV2.
```